src/components/common/CitiesList/CitiesList.jsx

```diff
@@ -1,21 +1,38 @@
-import CityCard from '../CityCard/CityCard';
+import { useMemo } from 'react';
 import { memo } from 'react';
+import CityCard from '../CityCard/CityCard';
+import usePopularCities from '../../../hooks/usePopularCities';
+import NotFoundSection from '../../../pages/CitiesPage/NotFoundSection';
+import CitiesListSkeleton from './CitiesListSkeleton';
+
+const CitiesList = ({ limit, increaseLimit, showLoadMoreButton }) => {
+  const { data: cities = [], error, isLoading } = usePopularCities();
 
-const CitiesList = ({ cities, limit, increaseLimit, showLoadMoreButton }) => {
   const defaultLimit = 10;
   const citiesLimit = limit || defaultLimit;
-  const displayedCities = cities.slice(0, citiesLimit);
-  const allCitiesDisplayed = citiesLimit >= cities.length;
+  const displayedCities = useMemo(
+    () => cities?.slice(0, citiesLimit) || [],
```

Comment:
you don't need optional chaining or fallback value here due to having a default value while destructing result of `usePopularCities` hook